### PR TITLE
Add Contributing Guidelines

### PR DESCRIPTION
**Pull request by sroopsai**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+3. **Set up remote upstream**: Add the original repository as a remote named "upstream" using `git remote add upstream [original repository URL]`.
+4. **Create a new branch**: Use `git checkout -b [branch-name]` to create a new branch for your contribution.
+
+## Making Changes
+
+1. **Make your changes**: Implement the changes or additions you want to make. Please follow any coding standards and guidelines provided in the project.
+2. **Test your changes**: Ensure that your changes work as expected and don't introduce any new issues. Run the following tests:
+   - **End-to-End (E2E) tests**: Execute E2E tests using the provided scripts. You can run E2E tests with `npm run test:e2e`. Make sure they pass before submitting your changes.
+   - **Regular integration tests**: Run regular integration tests to validate the functionality. You can run integration tests with `npm run test:integration`.
+3. **Update snapshots**: If you've made changes that affect snapshots, update them using `npm run test:e2e-update`.
```
Comment (dzikowski) [inline CONTRIBUTING.md:39]:
```suggestion
3. **Update snapshots**: If you've made changes that affect snapshots (esp. any template changes), update them using `npm run test:e2e-update`.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+2. **Clone your fork**: Use `git clone` to clone the repository to your local machine.
+3. **Set up remote upstream**: Add the original repository as a remote named "upstream" using `git remote add upstream [original repository URL]`.
+4. **Create a new branch**: Use `git checkout -b [branch-name]` to create a new branch for your contribution.
+
+## Making Changes
+
+1. **Make your changes**: Implement the changes or additions you want to make. Please follow any coding standards and guidelines provided in the project.
+2. **Test your changes**: Ensure that your changes work as expected and don't introduce any new issues. Run the following tests:
+   - **End-to-End (E2E) tests**: Execute E2E tests using the provided scripts. You can run E2E tests with `npm run test:e2e`. Make sure they pass before submitting your changes.
+   - **Regular integration tests**: Run regular integration tests to validate the functionality. You can run integration tests with `npm run test:integration`.
```
Comment (dzikowski) [inline CONTRIBUTING.md:38]:
```suggestion
   - **Unit tests**: Execute unit tests using the provided scripts. Use: `npm run test:unit`.
   - **End-to-End (E2E) tests**: Execute E2E tests using the provided scripts. Use: `npm run test:e2e`.
   - **Ent-to-End network tests**: Execute relevant shell scripts from `e2e-network` directory with E2E network tests.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+1. **Fork the repository**: Click the "Fork" button at the top right corner of the repository's page on GitHub.
+2. **Clone your fork**: Use `git clone` to clone the repository to your local machine.
+3. **Set up remote upstream**: Add the original repository as a remote named "upstream" using `git remote add upstream [original repository URL]`.
+4. **Create a new branch**: Use `git checkout -b [branch-name]` to create a new branch for your contribution.
+
+## Making Changes
+
+1. **Make your changes**: Implement the changes or additions you want to make. Please follow any coding standards and guidelines provided in the project.
+2. **Test your changes**: Ensure that your changes work as expected and don't introduce any new issues. Run the following tests:
```
Comment (dzikowski) [inline CONTRIBUTING.md:36]:
```suggestion
2. **Test your changes**: Ensure that your changes work as expected and don't introduce any new issues. Depending on the scope of your changes you may rely on our CI pipelines or run the following tests:
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+   - **End-to-End (E2E) tests**: Execute E2E tests using the provided scripts. You can run E2E tests with `npm run test:e2e`. Make sure they pass before submitting your changes.
+   - **Regular integration tests**: Run regular integration tests to validate the functionality. You can run integration tests with `npm run test:integration`.
+3. **Update snapshots**: If you've made changes that affect snapshots, update them using `npm run test:e2e-update`.
+
+## Submitting Changes
```
Comment (dzikowski) [inline CONTRIBUTING.md:41]:
```suggestion
## Running Fablo locally

You may want to verify some changes by running Fablo locally. To do so:
1. Execute `./fablo-build.sh` script to create a Fablo Docker image locally.
2. Use `./fablo.sh` from source root directory to call Fablo commands.

## Submitting Changes
```

**Review by dzikowski — COMMENTED** [review]
Thanks a lot @sroopsai!

I've added some suggestions related with testing. Fablo is quite heavy tool in terms of testing and we may rely at some extent on the CI. Also, I've added a section on how to run Fablo locally.

Once the changes are applied, I'll approve and merge.

File: CONTRIBUTING.md (hunk 4 above)
Comment (sroopsai) [reply CONTRIBUTING.md:48]: Done

File: CONTRIBUTING.md (hunk 3 above)
Comment (sroopsai) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 2 above)
Comment (sroopsai) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 1 above)
Comment (sroopsai) [reply CONTRIBUTING.md]: Done

**Review by dzikowski — APPROVED** [review]

**Pull request merged**